### DISCUSSION TITLE
Deploy into GitHub Pages for improved manageability

### DISCUSSION THREAD

**Discussion by awentzel** [microsoft/fast#5550] [Ideas]
## Executive Summary
Currently, our infrastructure depends on Azure Cloud, which requires special permissions for @fast-collaborators and @fast-contributors.  This discussion is intended to reach an agreement on migrating to GitHub Pages away from Azure Cloud.

### Requirements
* GitHub Pages supports static sites only

### Pros
* Leverage GitHub Environments to track deployment history, add required approvers, include delay deployments, and map directly to specific directories.
* Reduce Azure Cloud costs, by leveraging free GitHub Pages.
* Map to sub-directories instead of sub-domains 
* * Reducing DNS management tasks
* * Deploying to a single web service
* * Easier to maintain one service as opposed to one service per-site
* Eliminates RBAC requirements
* Eliminates infrastructure, removing the need to migrate to PME environment
* May require extending the development to production release branch policy
* Does not require SAW or PME Azure accounts
* Does not require Microsoft Internal support
* Could create multiple GitHub Environments

### Cons
* Less flexibility and control over infrastructure( reliability, scalability, application stack, etc)
* Requires CI/CD work to modify existing GH Actions to deploy to a new environment.
* Requires application and documentation code changes to build into a single site with sub-directories
* Requires documentation review to ensure all previous links are updated and old links maintained in permanent redirect file to inform search bots.
* Only supports static sites, so if any desire to extend functionality this may require Azure again
* May require a new release branch strategy

### Recommendation
Given the team's global distribution and collaborations both internally and externally, I'm recommending we leverage GitHub Pages to reduce costs and improve manageability while reducing internal compliance requirements.  This will require a little upfront work with a lot less back-end support which seems prudent at this time.

**Comment by EisenbergEffect** [comment]
This sounds like a great way to go. @awentzel Looks like most of the cons would affect your work as part of the effort. How do you feel about that?

**Reply by awentzel** [reply]
Waiting for after directory migration to determine go/no-go on this work.
